### ISSUE TITLE
Issue with Instagram Login when Providing a wrong or expired access_token

### ISSUE THREAD

**Issue by mpetyx** [pennersr/django-allauth#1888]
The problem is when trying to login with instagram, if for any reason the user's access_token is expired or non-valid the library returns http error 500 with the traceback.
Here is the traceback in discussion:
```
 File "/app/.heroku/python/lib/python2.7/site-packages/allauth/socialaccount/providers/instagram/views.py" in complete_login
  23.                                                              extra_data)

File "/app/.heroku/python/lib/python2.7/site-packages/allauth/socialaccount/providers/base.py" in sociallogin_from_response
  86.         uid = self.extract_uid(response)

File "/app/.heroku/python/lib/python2.7/site-packages/allauth/socialaccount/providers/instagram/provider.py" in extract_uid
  32.         return str(data['data']['id'])

Exception Type: KeyError at /v1/auth/instagram/
Exception Value: 'data'
Request information:
USER: AnonymousUser
```
I think this can be easily handled by raising an exception with a nice informative message in the extract_uid instead of sending this whole traceback.

**Comment by pennersr** [comment]
Closed via d4a2873f

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/1888
